lib/updateDataDisplay.ml

```diff
@@ -251,9 +246,13 @@ and eval_compound_var conf base env xx sl =
     | ("escape" | "html_encode") :: sl ->
         (Util.escape_html (loop sl) :> string) (* FIXME? *)
     | "safe" :: sl -> (Util.safe_html (loop sl) :> string) (* FIXME? *)
-    | [ "subs"; n; s ] ->
-        let n = int_of_string n in
-        if String.length s > n then String.sub s 0 (String.length s - n) else ""
+    | [ "subs"; n; s ] -> (
+        match int_of_string_opt n with
+        | Some n ->
+            if String.length s > n then String.sub s 0 (String.length s - n)
+            else (Printf.sprintf "String shorter that requested\n"
+              |> !GWPARAM.syslog `LOG_WARNING; s)
```

Comment:
here you don't need to bother with a Printf.sprintf and a pipe and just write:
```ocaml
!GWPARAM.syslog `LOG_WARNING "String shorter that requested\n" ;
s
```

thanks for adding logs!

Reply:
ok